### PR TITLE
Integration ecs

### PR DESCRIPTION
**Pull request by skumbananaaa**

### CODE REVIEW

File: LambdaEngine/Include/Rendering/RenderGraphTypes.h [hunk 1]
```diff
@@ -286,11 +294,13 @@ namespace LambdaEngine
...
-		String							ResourceName					= "";
-		String							RenderStageName					= "";
-		bool							Removable						= true;
-		ERenderGraphResourceBindingType BindingType						= ERenderGraphResourceBindingType::NONE;
-		int32							InputLinkIndex					= -1;
+		String							ResourceName		= "";
+		ERenderGraphResourceType		ResourceType		= ERenderGraphResourceType::NONE;
+		String							RenderStageName		= "";
+		bool							Removable			= true;
+		uint32							DrawArgsMask		= 0xFFFFFFFF;
```
Comment (TheoBerlin) [inline LambdaEngine/Include/Rendering/RenderGraphTypes.h:301]:
```suggestion
		uint32							DrawArgsMask		= UINT32_MAX;
```

File: LambdaEngine/Include/Resources/ResourceManager.h [hunk 2]
```diff
@@ -56,7 +58,7 @@ namespace LambdaEngine
 		*	result - A vector where all loaded GameObject(s) will be stored
```
Comment (norda97) [inline LambdaEngine/Include/Resources/ResourceManager.h]: Change comment to MeshComponents

File: LambdaEngine/Source/Engine/EngineLoop.cpp [hunk 3]
```diff
@@ -185,8 +185,8 @@ namespace LambdaEngine
 		{
 			return false;
 		}
-
-		if (!Renderer::Init())
+
```
Comment (TheoBerlin) [inline LambdaEngine/Source/Engine/EngineLoop.cpp:188]: :(

File: LambdaEngine/Source/Resources/ResourceLoader.cpp [hunk 4]
```diff
@@ -229,7 +229,7 @@ namespace LambdaEngine
 		TArray<Mesh*>		Meshes;
 		TArray<Material*>	Materials;
 		TArray<Texture*>	Textures;
-		TArray<GameObject>	LoadedGameObjects;
+		TArray<MeshComponent>	LoadedGameObjects;
```
Comment (norda97) [inline LambdaEngine/Source/Resources/ResourceLoader.cpp]: Change name of variable LoadedMeshComponenets for clarity

File: LambdaEngine/Source/Resources/ResourceLoader.cpp [hunk 5]
```diff
@@ -405,10 +405,10 @@ namespace LambdaEngine
 			{
 				context.Meshes.EmplaceBack(pNewMesh);
 
-				GameObject NewGameObject;
-				NewGameObject.Mesh		= context.Meshes.GetSize() - 1;
-				NewGameObject.Material	= context.MaterialIndices[pMesh->mMaterialIndex];
-				context.LoadedGameObjects.PushBack(NewGameObject);
+				MeshComponent newGameObject;
```
Comment (norda97) [inline LambdaEngine/Source/Resources/ResourceLoader.cpp]: Change name of variable

File: LambdaEngine/Source/Resources/ResourceManager.cpp [hunk 6]
```diff
@@ -49,9 +49,9 @@ namespace LambdaEngine
 		return true;
 	}
 
-	bool ResourceManager::LoadSceneFromFile(const String& filename, TArray<GameObject>& result)
+	bool ResourceManager::LoadSceneFromFile(const String& filename, TArray<MeshComponent>& result)
 	{
-		TArray<GameObject> sceneLocalGameObjects;
+		TArray<MeshComponent> sceneLocalGameObjects;
```
Comment (norda97) [inline LambdaEngine/Source/Resources/ResourceManager.cpp]: Change name of variable

**Review by norda97 — APPROVED** [review]

File: LambdaEngine/Source/Rendering/RenderGraphParser.cpp [hunk 7]
```diff
@@ -69,7 +69,8 @@ namespace LambdaEngine
 				renderStagesByName,
 				resourceStatesByHalfAttributeIndex,
 				resourceStateLinksByLinkIndex,
-				renderStageWeightsByName))
+				renderStageWeightsByName,
+				0xFFFFFFFF))
```
Comment (TheoBerlin) [inline LambdaEngine/Source/Rendering/RenderGraphParser.cpp:73]:
```suggestion
				UINT32_MAX))
```

**Review by TheoBerlin — APPROVED** [review]

**Comment by skumbananaaa** [comment]
Very much cool stuff

**Pull request merged**